src/plugins/robots/generic/simulator/colored_blob_perspective_camera_default_sensor.cpp

```diff
@@ -77,15 +77,15 @@ namespace argos {
                 * So that the origin is up-left, the i axis goes to the right, and the j axis goes down
                 */
                SInt32 unI =
-                  - m_cCamEntity.GetImagePxWidth() /
+                  static_cast<UInt32>(- m_cCamEntity.GetImagePxWidth() /
```

Comment:
Are these casts really necessary? Is it a warning that makes them necessary? Also, this cast is wrong because it loses the sign.

Reply:
This variable should be called `nI` (not `unI`) and the cast should be `static_cast<SInt32>`. Also, I think consideration should be given here to whether truncating the value is the correct option (foor, ceil, or round should probably be applied first).

Reply:
The previous behavior was truncation, which seemed to work OK, so a cast seems appropriate to me. If you are trying to map Real X,Y -> integer coordinates, then you would want everything between [0.0, 0.9999999) to map down to 0 right?